File: flutter/lib/core/benchmark/log_db_row_size.dart
import 'package:logger/logger.dart';

import '../config/di/config_di.dart';
import 'get_db_row_size.dart';

void logDbRowSize(Map<String, dynamic> row, {
  String name = '',
  String tag = 'db_row_size',
  bool log = true,
  Logger? logger,}) {
  final sizeInBytes = getRowSizeInBytes(row);
  final sizeInKB = getRowSizeInKB(row);

  logger ??= getIt<Logger>();

  if (log) {
    logger.d('$tag | Row size for $name: $sizeInBytes bytes (${sizeInKB.toStringAsFixed(2)} KB)');
  }
}

void logTotalDbRowSize(List<Map<String, dynamic>> rows, {
  String name = '',
  String tag = 'db_row_size',
  bool log = true,
  Logger? logger,}) {
  final totalSizeInBytes = rows.fold<int>(0, (sum, row) => sum + getRowSizeInBytes(row));
  final totalSizeInKB = totalSizeInBytes / 1024;

  logger ??= getIt<Logger>();

  if (log) {
    logger.d('$tag | Total row size for $name: $totalSizeInBytes bytes (${totalSizeInKB.toStringAsFixed(2)} KB)');
  }
}

File: react-native/src/core/benchmark/logDbRowSize.ts
import { getRowSizeInBytes, getRowSizeInKB } from './getDbRowSize';

export function logDbRowSize(
  row: Record<string, unknown>,
  opts: { name?: string; tag?: string; log?: boolean } = {},
): void {
  const { name = '', tag = 'db_row_size', log = true } = opts;
  const sizeBytes = getRowSizeInBytes(row);
  const sizeKB = getRowSizeInKB(row);
  if (log) {
    // eslint-disable-next-line no-console
    console.log(`${tag} | Row size for ${name}: ${sizeBytes} bytes (${sizeKB.toFixed(2)} KB)`);
  }
}

export function logTotalDbRowSize(
  rows: Array<Record<string, unknown>>,
  opts: { name?: string; tag?: string; log?: boolean } = {},
): void {
  const { name = '', tag = 'db_row_size', log = true } = opts;
  const totalBytes = rows.reduce((sum, row) => sum + getRowSizeInBytes(row), 0);
  const totalKB = totalBytes / 1024.0;
  if (log) {
    // eslint-disable-next-line no-console
    console.log(`${tag} | Total row size for ${name}: ${totalBytes} bytes (${totalKB.toFixed(2)} KB)`);
  }
}


